src/Umbraco.Core/Extensions/StringExtensions.cs

```diff
@@ -390,7 +390,7 @@ public static string EnsureStartsWith(this string input, string toStartWith)
             return input;
         }
 
-        return toStartWith + input.TrimStart(toStartWith);
+        return toStartWith + input.TrimStartExact(toStartWith);
```

Comment:
This also seems a bit weird: it adds the `toStartWith`, but also (case-invariantly) trims it from the start 🙃

Together with adding the `StringComparison` parameter, this could become:
```c#
public static string EnsureStartsWith(this string input, string toStartWith, StringComparison comparisonType = default)
    => input.StartsWith(toStartWith, comparisonType) ? input : toStartWith + input;
```

And similarly for `EnsureEndsWith()`:
```c#
public static string EnsureEndsWith(this string input, string toEndWith, StringComparison comparisonType = default)
    => input.EndsWith(toEndWith, comparisonType) ? input : input + toEndWith;
```

Reply:
Sure it's wierd, but I think this belongs in a separate PR 😁